### ISSUE TITLE
Fix accessibility of the snippet editor help text

### ISSUE THREAD

**Issue by afercia** [Yoast/yoast-components#541]
The snippet editor help text introduced in #517 has room for improvements:

1
I understand the will to have an animation, however links within an element with height `0px` and `aria-hidden="true"` can still receive focus. Just use the keyboard or a screen reader + keyboard to test: there's one tab stop (the link in the help text) and nothing gets announced because of `aria-hidden="true"`. It's basically an "empty" tab stop. At the end of the animation, the element should be really hidden with `display: none`.

2
the toggle button has an `aria-label={ "yoast-help-text" }` which I guess is a development leftover. `aria-label` needs a meaningful text. Most importantly, since the help text may refer to different context, this text should be passed via prop to adapt to the different context. In the current metabox, for example, we're doing:
`Show information about the snippet editor`
`Show information about the focus keyword`
etc.

3
`aria-controls={ this.props.className + "__helpDiv" }`
`aria-controls` needs an ID that must be unique, not a className 

4
the link in the help text uses target=_blank but doesn't inform users it's going to open in a new tab. We have tools for this, for example `makeOutboundLink()`

5
Minor, related to styling: why the help text uses Arial? Ideally, it should inherit the font face from the context. This font stack doesn't look ideal to me, unless I'm missing something:
```
const YoastHelpText = styled.p`
	font-family: Arial, Helvetica, sans-serif;
	font-size: 13px;
`;
```

**Comment by afercia** [comment]
Also noticed that, since the default className is `className: "yoast-help-button"` things like `className={ this.props.className + "__yoast-help-button" }` will result in something like `yoast-help-button__yoast-help-button`.

**Comment by afercia** [comment]
I've pushed two branches, with and without CSS animation:
`stories/af/541-help-text-accessibility-no-animation`
`stories/af/541-help-text-accessibility-with-animation`

I'd need some feedback about which is the best option:
- no animation: is simpler: the help text is conditionally rendered so when collapsed it's not in the DOM
- with animation: more complex and and a bit buggy: the animation works when clicking normally on the toggle button. At the end of the closing animation, the help text is completely hidden with `display: none` and that fixes the a11y part. However, the animation doesn't work correctly in some edge cases and I wasn't able to fully understand why. To reproduce:
  - focus the button with the keyboard (just because using the keyboard we can activate the button faster)
  - press Enter very quickly 4 times
  - at this point, in most of the cases the element is collapsed but its display property is `block`, while it should be `none`

![screen shot 2018-06-04 at 11 27 55](https://user-images.githubusercontent.com/1682452/40910793-5e54388a-67ed-11e8-97f0-d5c695d02e50.jpg)

My guess is this has something to do with the `transitionend` event not occurring when clicking the button so fast. I'm certainly doing something wrong but that's a bit hard to track down. Overall, I think this could be a good starting point to implement a `slideToggle` component in the future, but I think it's not ready for production. Any help would be appreciated 🙂
